app/queries/reservations/active_on_date_query.rb

```diff
@@ -1,5 +1,5 @@
 module Reservations
-  class ReservedOnDateQuery < Reservations::ReservationsQueryBase
+  class ActiveOnDateQuery < Reservations::ReservationsQueryBase
```

Comment:
Isn't this basically `overlap_with_date(date).active`? I'm not sure it needs a separate Query Object, now that I think about it.